plugins/basic/nexus-repository-docker/src/main/java/org/sonatype/nexus/repository/docker/internal/DockerConnectorFacetImpl.java

```diff
@@ -0,0 +1,281 @@
+                                  final ConnectorRegistrar connectorRegistrar)
+  {
+    this.repositoryManager = checkNotNull(repositoryManager);
+    this.constraintViolationFactory = checkNotNull(constraintViolationFactory);
+    this.connectorRegistrar = checkNotNull(connectorRegistrar);
+  }
+
+  @Override
+  protected void doValidate(final Configuration configuration) throws Exception {
+    facet(ConfigurationFacet.class).validateSection(configuration, CONFIG_KEY, Config.class);
+
+    Config configToValidate = facet(ConfigurationFacet.class).readSection(configuration, CONFIG_KEY, Config.class);
+    Set<ConstraintViolation<?>> violations = new HashSet<>();
+    maybeAdd(violations, validatePortNotInUse(pathOf("httpPort"), configToValidate.httpPort));
+    maybeAdd(violations, validatePortNotInUse(pathOf("httpsPort"), configToValidate.httpsPort));
+    maybeAdd(violations, validateNotSamePort(
+        pathOf("httpPort"), configToValidate.httpPort, pathOf("httpsPort"), configToValidate.httpsPort
+    ));
+    ConstraintViolations.maybePropagate(violations, log);
+  }
+
+  @Override
+  protected void doConfigure(final Configuration configuration) throws Exception {
+    config = facet(ConfigurationFacet.class).readSection(configuration, CONFIG_KEY, Config.class);
+    log.debug("Config: {}", config);
+  }
+
+  @Override
+  protected void doStart() throws Exception {
+    if (config.httpPort != null) {
+      try {
+        httpConfig = new DockerConnectorConfiguration(getRepository().getName(), HttpScheme.HTTP, config.httpPort);
+        connectorRegistrar.addConnector(httpConfig);
+      }
+      catch (Exception e) {
+        log.warn(
+            "Could not configure HTTP connector on port {} for docker repository {}",
+            config.httpPort, getRepository().getName(), e
+        );
+      }
+    }
+    if (config.httpsPort != null) {
+      try {
+        httpsConfig = new DockerConnectorConfiguration(getRepository().getName(), HttpScheme.HTTPS, config.httpsPort);
+        connectorRegistrar.addConnector(httpsConfig);
+      }
+      catch (Exception e) {
+        log.warn(
+            "Could not configure HTTPS connector on port {} for docker repository {}",
+            config.httpsPort, getRepository().getName(), e
+        );
+      }
+    }
+  }
+
+  @Override
+  protected void doStop() throws Exception {
+    if (httpConfig != null) {
+      connectorRegistrar.removeConnector(httpConfig);
+    }
+    if (httpsConfig != null) {
+      connectorRegistrar.removeConnector(httpsConfig);
+    }
+  }
+
+  @Override
+  public Integer getHttpPort() {
+    return config.httpPort;
+  }
+
+  @Override
+  public Integer getHttpsPort() {
+    return config.httpsPort;
+  }
+
+  /**
+   * Validate that port is not used as used as HTTP/HTTPS ports in another repository.
+   */
+  private ConstraintViolation<?> validatePortNotInUse(final String path, final Integer port) {
```

Comment:
Could consider `ConnectorRegistrar#unavailablePorts()` too? This one validates that only docker reposes will not conflict, but it still may conflict with some other service using `ConnectorRegistrar`.